src/models/thread.ts

```diff
@@ -514,6 +514,15 @@ export class Thread extends ReadReceipt<EmittedEvents, EventHandlerMap> {
 
     public hasUserReadEvent(userId: string, eventId: string): boolean {
         if (userId === this.client.getUserId()) {
+
+            // We consider all threaded events read if they are part of a thread
+            // that has no activity since the first ever threaded event recorded in that room
+            // This prevents rooms to generated unwanted notifications for threads
+            // created before MSC3771
```

Comment:
I'm finding this comment very difficult to read, maybe the following is a bit clearer:

```suggestion

            // We consider all events in a thread as read if the thread has not had
            // activity since the first ever threaded receipt recorded in that room.
            // This prevents rooms generating unwanted notifications for threads
            // created before MSC3771.
```